firmware/raspberry_pi4_network_boot_beta.md

```diff
@@ -0,0 +1,119 @@
+sudo rpi-update
+
+# (Optional) Update rpi-eeprom-update to automatically get updates / bug fixes.
+sudo echo FIRMWARE_RELEASE_STATUS="beta" > /etc/default/rpi-eeprom-update
+```
+
+### Configuration - Enable network boot
+Network boot is not enabled by default in the bootloader. To enable it the bootloader configuration file must be edited.
+```
+# Extract the configuration file
+cp /lib/firmware/raspberrypi/beta/pieeprom-2019-09-22.bin pieeprom.bin
+rpi-eeprom-config pieeprom.bin > bootconf.txt
+
+# Enable network boot
+# change boot_order from 0x1 (sd-boot) to 
+BOOT_ORDER=0x21
+# sd-boot then failover to network boot
+
+# If you have a UART cable then setting BOOT_UART=1 will help debug network boot issues
+BOOT_UART=1
+
+# Apply the configuration change to the EEPROM image file
+rpi-eeprom-config --out pieeprom-netboot.bin --config bootconf.txt pieeprom.bin
+```
+
+### Update the bootloader EEPROM
+```
+# Flash the bootloader EEPROM with network boot enabled
+# Run 'rpi-eeprom-update -h' for more information
+sudo rpi-eeprom-update -d -f ./pieeprom-netboot.bin
+sudo reboot
+```
+
+## TFTP prefix
+As with the Pi3 bootloader the presence of /serial_number/start.elf is used to detect
+whether the serial number should be used as the directory prefix for TFTP requests.
+
+The bootloader loads start4.elf in preference to start.elf so it's possible to
+just have an empty start.elf file if you only want to have Pi4 firmware in the TFTP
+root directory.
+
+## Bootloader configuration
+
+### BOOT_ORDER
+The BOOT_ORDER setting allows flexible configuration for the priority of different
+bootmodes. It is represented as 32bit unsigned integer where each nybble represents
+a bootmode. The bootmodes are attempted in LSB to MSB order.  
+
+E.g. 0x21 means try SD first followed by network boot then stop.
+
+The retry counters are reset when switching to the next boot mode.
+
+BOOT_ORDER fields  
+* 0x0 - NONE (stop with error pattern)  
+* 0x1 - SD CARD  
+* 0x2 - NETWORK  
+
+Default: 0x00000001 (with 3 SD boot retries to match the current bootloader behaviour)  
+
+### SD_BOOT_MAX_RETRIES
+Specify the maximum number of times that the bootloader will retry booting from the sd-card.  
+-1 means infinite retries  
+Default: 0  
+
+### NET_BOOT_MAX_RETRIES
+Specify the maximum number of times that the bootloader will retry network boot.  
+-1 means infinite retries  
+Default: 0  
+
+### DHCP_TIMEOUT
+The timeout in milliseconds for the entire DHCP sequence before failing the current iteration.  
+Default: 45000  
+Minimum: 5000  
+
+### DHCP_REQ_TIMEOUT
+The timeout in milliseconds before retrying DHCP DISCOVER or DHCP REQ.  
+Default: 4000  
+Minimum: 500  
+
+### TFTP_TIMEOUT
+The timeout in milliseconds for an individual file download via TFTP.  
+Default: 15000  
+Minimum: 5000  
+
+### TFTP_IP
+Optional dotted decimal ip address for the TFTP server which overrides the server-ip from the DHCP request.  
```

Comment:
ip -> IP

Also, might be useful to give an example of what a dotted decimal IP address is - presumably that's one like 192.168.1.250 ?